src/traces/scattergl/index.js

```diff
@@ -259,25 +259,56 @@ function sceneUpdate(gd, subplot) {
         // draw traces in proper order
         scene.draw = function draw() {
             var i;
+            var we_must_draw_previous_line_marker = 0;
             for(i = 0; i < scene.count; i++) {
+                var we_defer_somes_draw = 0;
+                if(scene.fill2d && scene.fillOptions[i] && scene.fillOptions[i].fillmode && scene.fillOptions[i].fillmode === 'tonext') {
```

Comment:
I think you got the fill part right :tada: 

The logic is a little hard to follow; it would be nice to to :recycle: the logic from the SVG version:

https://github.com/plotly/plotly.js/blob/88d48c57ea7e9876fce134a6731ff7542178e3a8/src/traces/scatter/plot.js#L95-L105

by using `_nexttrace` and `_prevtrace` which are defined during [`linkTraces`](https://github.com/plotly/plotly.js/blob/master/src/traces/scatter/link_traces.js) called in scattergl [here](https://github.com/plotly/plotly.js/blob/88d48c57ea7e9876fce134a6731ff7542178e3a8/src/traces/scattergl/index.js#L369.

Moreover, looks like the error bars are drawn above the lines, SVG scatter does it the other way around.

----

Oh well, I'll try out a few things and hopefully that one or two commits will be enough to bring this to the finish line.

Thanks again.

Reply:
@ErwanMAS I made two commits on branch [`etienne-scattergl-ordering`](https://github.com/plotly/plotly.js/compare/etienne-scattergl-ordering) which appear to make things work while reusing the SVG scatter logic, See:

- https://github.com/plotly/plotly.js/commit/075bd145e9122a87456cf511d729b2e05e45859a
- https://github.com/plotly/plotly.js/commit/6ae6283657c6293640bedde0d7cfa1748320a35d

Do they look ok to you? If so, I'll release them along with your work in next week's `v1.42.0`.

Reply:
@etpinard for me this seem to be good .  Can you take the last version of https://github.com/plotly/plotly.js/blob/4b2b0d5d34e648e361532fe23f5dbff536792e1c/test/image/mocks/gl2d_scatter_fill_self_next_vs_nogl.json . The new version has now 4 charts .